drivers/interrupt_controller/Kconfig.xlnx

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2023-2024 Advanced Micro Devices, Inc. (AMD)
+# Copyright (c) 2023 Alp Sayin <alpsayin@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config XLNX_INTC
+	bool "Xilinx interrupt controller"
+	help
+	  The AXI Interrupt Controller (INTC).
+
+menu "AXI Interrupt Controller Optional Registers"
+	depends on XLNX_INTC
+
+	config XLNX_INTC_USE_IPR
+	bool "Use Interrupt Pending Register"
+	help
+	  Each bit in this register is the logical AND of the bits in the ISR and the IER.
+	  This is an optional read-only register. Don't choose to use it if it doesn't exist.
+
+	config XLNX_INTC_USE_SIE
+	bool "Use Set Interrupt Enables Register"
+	help
+	  Writing a 1 to a bit location in SIE sets the corresponding bit in the IER.
+	  This is an optional write-only register. Don't choose to use it if it doesn't exist.
+
+	config XLNX_INTC_USE_CIE
+	bool "Use Clear Interrupt Enables Register"
+	help
+	  Writing a 1 to a bit location in CIE clears the corresponding bit in the IER.
+	  This is an optional write-only register. Don't choose to use it if it doesn't exist.
+
+	config XLNX_INTC_USE_IVR
+	bool "Use Interrupt Vector Register"
+	help
+	  The IVR contains the ordinal value of the highest priority, enabled, and active interrupt input.
+	  The IVR acts as an index to the correct Interrupt Vector Address.
+	  This is an optional read-only register. Don't choose to use it if it doesn't exist.
+
+	config XLNX_INTC_INITIALIZE_IVAR_REGISTERS
+	bool "Initialize Interrupt Vector Address Registers"
+	help
+	  The IVAR contains the addresses for fast-interrupts. This flag enables initializing all
+	  the address registers to point them to default interrupt handler which is `0x10` for Microblaze.
```

Comment:
These sound more like hardware description? Should they be devicetree properties instead?

Reply:
I agree with Henrik, configuring hardware features or behavior via the respective driver's Kconfig hasn't been the usual procedure any more for quite some time now. This should rather be a hand full of bools in the driver's devicetree binding.

Also, the Kconfig approach comes with a major limitation: it only works properly for devices with an instance count of 1, as is usually the case with interrupt controllers (I never had this one in one of my block designs before, but I guess with an IP-based controller, there could be more than one instance?). Unless you include all supported settings for all possible instances (e.g. GEM0-GEM3 for the UltraScale's GEMs) into the Kconfig, which comes with a lot of duplication in the Kconfig file, you can't have per-device settings this way. And with IP-core based devices (e.g. devices like AXI GPIO), you might not even have a fixed number of device instances, i.e. you couldn't provide per-device settings for any possible instance of the device.

Reply:
Agree on this one.